vicinity/integrations/huggingface.py

```diff
@@ -4,135 +4,130 @@
-            }
-            config_path = temp_path / "config.json"
-            config_path.write_text(json.dumps(config))
-            upload_file(
-                repo_id=repo_id,
-                path_or_fileobj=config_path,
-                token=token,
-                repo_type="dataset",
-                path_in_repo="config.json",
-            )
-
-        # Load the dataset card template from the related path
-        template_path = Path(__file__).parent / "dataset_card_template.md"
-        template = template_path.read_text()
-        content = template.format(repo_id=repo_id, num_items=len(self.items), config=json.dumps(config, indent=4))
-        return DatasetCard(content=content).push_to_hub(repo_id=repo_id, token=token, repo_type="dataset")
-
-    @classmethod
-    def load_from_hub(cls, repo_id: str, token: str | None = None, **kwargs: Any) -> "Vicinity":
-        """
-        Load a Vicinity instance from the Hugging Face Hub.
-
-        :param repo_id: The repository ID on the Hugging Face Hub.
-        :param token: Optional authentication token for private repositories.
-        :param **kwargs: Additional arguments passed to load_dataset.
-        :return: A Vicinity instance loaded from the Hub.
-        """
-        try:
-            from datasets import load_dataset
-            from huggingface_hub import snapshot_download
-        except ImportError:
-            raise _HUB_IMPORT_ERROR
-
-        # Load dataset and extract items and vectors
-        dataset = load_dataset(repo_id, token=token, split="train", **kwargs)
-        if "items" in dataset.column_names:
-            items = dataset["items"]
-        else:
-            # Create items from all columns except 'vectors'
-            items = []
-            columns = [col for col in dataset.column_names if col != "vectors"]
-            for i in range(len(dataset)):
-                items.append({col: dataset[col][i] for col in columns})
-        has_vectors = "vectors" in dataset.column_names
-        vector_store = BasicVectorStore(vectors=dataset["vectors"]) if has_vectors else None
-
-        # Download and load config and backend
-        repo_path = Path(snapshot_download(repo_id=repo_id, token=token, repo_type="dataset"))
-        with open(repo_path / "config.json") as f:
-            config = json.load(f)
-            model_name_or_path = config.pop("model_name_or_path")
-
-        print(_MODEL_NAME_OR_PATH_PRINT_STATEMENT.format(model_name_or_path=model_name_or_path))
-        backend_type = Backend(config["backend_type"])
-        backend = get_backend_class(backend_type).load(repo_path / "backend")
-
-        return cls(items=items, backend=backend, metadata=config["metadata"], vector_store=vector_store)
+def push_to_hub(
+    repo_id: str,
+    items: list[Any],
+    backend: AbstractBackend,
+    metadata: dict[str, str],
+    vector_store: BasicVectorStore | None = None,
+    model_name_or_path: str | None = None,
```

Comment:
I don't see why we have this as a separate parameter. Couldn't this just be a metadata key?